Model/Api/CreateOrder.php

```diff
@@ -680,7 +680,7 @@ public function validateShippingCost($quote, $transaction)
         }
         if ($quote->getShippingAddress() && !$quote->isVirtual()) {
             $amount = $quote->getShippingAddress()->getShippingAmount();
-            if (! $this->isBackOfficeOrder($quote)) {
+            if (! $this->isBackOfficeOrder($quote) && !$this->cartHelper->ignoreAdjustingShippingAmount($quote)) {
```

Comment:
So you mean we do not deduct from shipping amount shipping discount if it has type DISCOUNT_ACTION_FIXED_AMOUNT_FOR_CART. But what about other types? In which cases do we still need to deduct?
I think we need to remove this adjustment completely or have an integration test that covers this adjustment.

Reply:
In which cases do we still need to deduct?
=> @vitaliyreznikov I tested and confirmed that we still need to deduct for cases DISCOUNT_ACTION_BY_PERCENT, DISCOUNT_ACTION_FIXED_AMOUNT. @dumega Can you confirm these since you added the deducting logic

Reply:
@johnmccombs1 This logic is not added by me, and was added when creating this file for Bolt pre-auth (https://github.com/BoltApp/bolt-magento2/commit/8134b200197201a76a660663b84984baecad9e07#diff-208413ae52687204accddd79663f92e3a8277b5643fe0e935b16607e4ad9511e).  @danac-gs  any insight on this?

Reply:
> any insight on this?
Not really. I think the case John is covering (apply discount to shipping) was deliberately skipped because it was assumed not needed / urgent at a time so it was left as is. I know from doing tests that that path was not working in all cases but seems like no merchants asked for it so far. Anyway, I believe John's approach is OK if it passes all test cases.